core/jobs.py

```diff
@@ -1657,6 +1724,32 @@ def on_batch_job_failure(cls):
             cls._kickoff_batch_job_after_previous_one_ends()
 
 
+def _deep_convert_to_bytes(value):
```

Comment:
Some jobs already do .encode('utf-8') or similar within the job. Do those need to be fixed?

Reply:
Nope, although many test outputs need to be updated unfortunately.

Reply:
Just to check ... why don't they need to be fixed?

Reply:
Ah, because utf-8 encoding an already utf-8 encoded string is a no-op.

Specifically, utf-8 encodings takes a code-point like 🙂 and transforms it into a sequence of bytes like `'\xf0\x9f\x99\x82'`. Such bytes are not changed to something else by a utf-8 encoder, so it will remain the same.

Reply:
-- ignore last comment, need to do more research.

Reply:
OK, np. Please reassign me once you'd like me to take another look at this PR.